packages/terra-switch/src/Switch.jsx

```diff
@@ -0,0 +1,152 @@
+import React, { useRef, useCallback } from 'react';
+import PropTypes from 'prop-types';
+import classNames from 'classnames';
+import classNamesBind from 'classnames/bind';
+import ThemeContext from 'terra-theme-context';
+import { FormattedMessage } from 'react-intl';
+import { KEY_SPACE, KEY_RETURN } from 'keycode-js';
+import { enableFocusStyles, removeFocusStyles, restoreFocusStyles } from './_SwitchUtils';
+import styles from './Switch.module.scss';
+
+const cx = classNamesBind.bind(styles);
+
+const SWITCH_STATE = Object.freeze({
+  ON: <FormattedMessage id="Terra.switch.switchstatuslabel.on" />,
+  OFF: <FormattedMessage id="Terra.switch.switchstatuslabel.off" />,
+});
+
+const propTypes = {
+  /**
+   * Id of the Switch button.
+   */
+  switchId: PropTypes.string.isRequired,
+  /**
+   * Whether or not the Switch is checked ("ON").
+   */
+  isChecked: PropTypes.bool,
+  /**
+   * Whether or not the Switch is disabled.
+   */
+  isDisabled: PropTypes.bool,
+  /**
+   * The label text of the Switch component.
+   */
+  labelText: PropTypes.string.isRequired,
+  /**
+   * Callback function when switch value changes from ON / OFF.
+   * Returns Parameters: 1. switch value 2. event.
+   */
+  onChange: PropTypes.func,
+};
+
+const defaultProps = {
+  isChecked: false,
+  isDisabled: false,
+  onChange: undefined,
+};
+
+const Switch = (props) => {
+  const {
+    switchId,
+    isChecked,
+    isDisabled,
+    onChange,
+    labelText,
+    ...customProps
+  } = props;
+  const theme = React.useContext(ThemeContext);
+  const sliderButton = useRef();
+
+  const handleOnClick = useCallback((event) => {
+    // See https://developer.mozilla.org/en-US/docs/Web/HTML/Element/Button#Clicking_and_focus
+    // Button on Firefox, Safari and IE running on OS X does not receive focus when clicked.
+    // This will set focus on the button when clicked.
+    sliderButton.current.focus();
+    if (onChange) {
+      onChange(!isChecked, event);
+    }
+  }, [isChecked, onChange]);
+
+  const handleOnKeyDown = (event) => {
+    if (event.nativeEvent.keyCode === KEY_RETURN || event.nativeEvent.keyCode === KEY_SPACE) {
+      event.preventDefault();
+      enableFocusStyles(sliderButton.current);
+      if (onChange) {
+        onChange(!isChecked, event);
+      }
+    }
+  };
+
+  const handleOnMouseDown = (event) => {
+    event.preventDefault();
+    removeFocusStyles(sliderButton.current);
+  };
+
+  const handleOnBlur = () => {
+    restoreFocusStyles(sliderButton.current);
+  };
+
+  const labelClassNames = cx([
+    'label-text',
+    { 'is-disabled': isDisabled },
```

Comment:
Now that there is a single entry point for the control we can remove the individual `is-disabled` classes from the sub-elements and instead with child/nested selectors from the parent tray's `is-disabled` class.

Reply:
but Since now we have changed button to div disabled is not happening as it was happening before when button was wrapped around all other controls.  sub-elements do have theme variables for disabled styles like tray has theme variables for background-color, box-shadow and border styles. So removing these classes will enable focus styles of slider on TAB key press and hover styles on mouse hover, even when switch is disabled.